kubectl-plugin/pkg/cmd/job/job_submit.go

```diff
@@ -489,15 +496,14 @@ func (options *SubmitJobOptions) Run(ctx context.Context, factory cmdutil.Factor
 	if rayJobID == "" {
 		rayJobID = <-rayJobIDChan
 	}
-	// Add annotation to RayJob with the correct Ray job ID and update the CR
-	options.RayJob, err = k8sClients.RayClient().RayV1().RayJobs(*options.configFlags.Namespace).Get(ctx, options.RayJob.GetName(), v1.GetOptions{})
+
+	// Patch the ray job with the correct Ray job ID.
+	patch := []jsonpatch.Operation{jsonpatch.NewOperation("add", "/spec/jobId", rayJobID)}
+	raw, err := json.Marshal(patch)
 	if err != nil {
-		return fmt.Errorf("Failed to get latest version of Ray job")
+		return fmt.Errorf("Generate Ray Job ID patch: %w", err)
 	}
-
-	options.RayJob.Spec.JobId = rayJobID
-
-	_, err = k8sClients.RayClient().RayV1().RayJobs(*options.configFlags.Namespace).Update(ctx, options.RayJob, v1.UpdateOptions{})
+	_, err = k8sClients.RayClient().RayV1().RayJobs(*options.configFlags.Namespace).Patch(ctx, options.RayJob.Name, types.JSONPatchType, raw, v1.PatchOptions{})
```

Comment:
Better to keep using Update as [I learned here](https://github.com/ray-project/kuberay/pull/2926#discussion_r1944049984)?

Reply:
+1

Reply:
in this case

1. Update fails, see commit message
2. I don't expect data races
3. If there was a write race, the consequence would be small - I think this field is largely informational?

Regardless, it's simple to add a resource version check. I will upload a second PR.